pkgs/tools/misc/bc/default.nix

```diff
@@ -9,9 +9,17 @@ stdenv.mkDerivation rec {
 
   configureFlags = [ "--with-readline" ];
 
-  buildInputs = [flex readline ed texinfo];
+  # As of 1.07 cross-compilation is quite complicated as the build system wants
+  # to build a code generator, bc/fbc, on the build machine.
+  patches = [ ./cross-bc.patch ];
+  nativeBuildInputs = [autoreconfHook flex ed.out texinfo] ++
+    stdenv.lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform)
+    [buildPackages.stdenv.cc buildPackages.readline.out buildPackages.ncurses.out];
```

Comment:
BuildPackages.stdenv.cc should be a depsBuildBuild, can be unconditional too.

Reply:
The other two can be made unconditional too, based on the env vars below. Maybe add a comment saying for cross too since people might be confused without the conditional and remove it.